### DISCUSSION TITLE
Postgres 14 ARRAY.Comparator.any and contains/like

### DISCUSSION THREAD

**Discussion by rsaleev** [sqlalchemy/sqlalchemy#10587] [Usage Questions]
### Describe the bug

Can't provide proper function and operator to achieve comparison of each item with string

```
@pytest.mark.parametrize(argnames='phone', argvalues=['+79001234567','9001234567'])
async def test_client_db_contacts(plain_db_session, phone):
    stmt = select(ClientContactsNotification).filter(any_(ClientContactsNotification.phones).contains(phone))
    result = await plain_db_session.execute(stmt)
 ```
throws ```syntax error at or near "ANY"```

```
@pytest.mark.parametrize(argnames='phone', argvalues=['+79001234567','9001234567''])
async def test_client_db_contacts(plain_db_session, phone):
    stmt = select(ClientContactsNotification).filter(ClientContactsNotification.phones.contains(phone))
    result = await plain_db_session.execute(stmt)
```
throws ```operator does not exist: character varying[] @> character varying```

proper way in plain SQL to achieve it without speacial array functions
```
select * from delivery.client_contacts_notification ccn 
where array_to_string(ccn.phones, ';') like '%1234567%';
```

### Optional link from https://docs.sqlalchemy.org which documents the behavior that is expected

https://docs.sqlalchemy.org/en/20/core/type_basics.html#sqlalchemy.types.ARRAY.Comparator.contains

### SQLAlchemy Version in Use

2.0.2

### DBAPI (i.e. the database driver)

asyncpg

### Database Vendor and Major Version

PostgreSQL 14

### Python Version

3.10

### Operating system

Linux

### To Reproduce

```python
from sqlalchemy.dialects.postgresql import ARRAY
from sqlalchemy.orm import Mapped, mapped_column
from sqlalchemy import String, Integer
from sqlalchemy.orm import DeclarativeBase

class BaseModel(DeclarativeBase):
    pass

class ClientContactsNotification(BaseModel):
    __tablename__ = "client_contacts_notification"
    __table_args__ = {"schema": "delivery"}

    id: Mapped[int] = mapped_column(Integer, primary_key=True)
    id_post_operation: Mapped[int] = mapped_column(
        ForeignKey("delivery.post_operation.id")
    )
    phones: Mapped[List[str]] = mapped_column(ARRAY(String))
    emails: Mapped[List[str]] = mapped_column(ARRAY(String))
    identifiers: Mapped[List[str]] = mapped_column(ARRAY(String))
```

### Error

```
# Copy the complete stack trace and error message here, including SQL log output if applicable.
[undefined]sqlalchemy.exc.ProgrammingError: (sqlalchemy.dialects.postgresql.asyncpg.ProgrammingError) <class 'asyncpg.exceptions.UndefinedFunctionError'>: operator does not exist: character varying[] @> character varying
HINT:  No operator matches the given name and argument types. You might need to add explicit type casts.
[SQL: SELECT delivery.client_contacts_notification.id, delivery.client_contacts_notification.id_post_operation, delivery.client_contacts_notification.phones, delivery.client_contacts_notification.emails, delivery.client_contacts_notification.identifiers 
FROM delivery.client_contacts_notification 
WHERE delivery.client_contacts_notification.phones @> $1::VARCHAR]
```

**Comment by CaselIT** [comment]
Hi,

It's not clear to me what sql you want to execute. Can you provide what sql you expect yo execute?

**Comment by rsaleev** [comment]
> Hi,
> 
> It's not clear to me what sql you want to execute. Can you provide what sql you expect yo execute?

I want to compare array's element to parameter (string) either with `ilike` operator or `contains`. It will be my filter for select statement.
This query works fine, because all arrays contains only 1 item, so I can use it 
```select * from delivery.client_contacts_notification ccn 
where array_to_string(ccn.phones, ',') like '%222298749%';```

**Comment by CaselIT** [comment]
> I want to compare array's element to parameter (string) either with ilike operator or contains

sorry, it's still not clear to me what sql you want to execute here.

> https://docs.sqlalchemy.org/en/20/core/type_basics.html#sqlalchemy.types.ARRAY.Comparator.contains 

this is a bug in the documentation, it should not be there

**Reply by CaselIT** [reply]
the issue in the docs will be fixed here https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4961

**Comment by CaselIT** [comment]
Moved to discussion. Ensure you are using the array type imported from the postgresql dialect and contains should work
